clang/lib/AST/DeclPrinter.cpp

```diff
@@ -129,11 +118,13 @@ namespace {
                                 const TemplateParameterList *Params);
     void printTemplateArguments(llvm::ArrayRef<TemplateArgumentLoc> Args,
                                 const TemplateParameterList *Params);
-
-    inline void prettyPrintAttributes(Decl *D) {
-      prettyPrintAttributes(D, Out);
-    }
-
+    enum class AttrPosAsWritten {
+      Unknown = 0,
```

Comment:
'Unknown' is likely to happen with 'implicit' attributes.  Perhaps the 'CanPrintOnLeft' is worth keeping so that we can aggressively print those on the left?

Reply:
Unknown means that we have no preference and we want to keep the old behavior of `prettyPrintAttributes`.